migration/migrator/migrator.go

```diff
@@ -69,14 +85,17 @@ type Migrator struct {
 	// maps between semantic version to the timestamp it was created at
 	versionMapper map[string]uint
 	versions      version.Collection
+
+	postHook func(context.Context) error
```

Comment:
Currently the `c.dialectExecutor.Finalize` is used as the postHook, because there can be multiple things the cq migrator can "go through" and after all of them the dialect finalizer has to be called to clean up history. Calling it on drop currently is harmless, and it's internal to cq-core and SDK anyway, so not really a provider exposed thing.